kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h

```diff
@@ -1125,17 +1136,14 @@ class ResidualBasedEliminationBuilderAndSolverWithConstraints
         SparseMatrixMultiplicationUtility::MatrixMultiplication(T_transpose_matrix, rA, auxiliar_A_matrix);
 
         // We do a backup of the matrix before apply the constraints
-        if (mpOldAMatrix == NULL) { // If the pointer is not initialized initialize it to an empty matrix
-            TSystemMatrixPointerType pNewOldAMatrix = TSystemMatrixPointerType(new TSystemMatrixType(0, 0));
-            mpOldAMatrix.swap(pNewOldAMatrix);
-        }
-        (*mpOldAMatrix).swap(rA);
+        (mOldAMatrix).swap(rA);
```

Comment:
take care, it is not any longer a pointer... this operation is now expensive (and furthermore not compatible with trilinos)

Reply:
I will revert this. But on the Block Builder and solver C and T are not pointers

Reply:
you are right about the Block version. I should change it there too